### PR TITLE
Avoid error using only local translation files

### LINKED ISSUE #298: [Bug] Frontend fails to run if translations aren't fetched first
## Description
Sometimes new translations will be available for the frontend but if you don't run `i18n:get-translations` before running `npm run dev` the API will crash after building.

## Reproduction

1. Checkout the repository and delete a translation `.json` file
2. Run `npm run dev`
3. See error about missing translation files

## Expectation
Missing translations shouldn't crash the server

## Additional context
Expectation according to @zackkrida 

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Fixes #298 by @sarayourfriend

## Description

This addresses the problem of the project failing to run if not all translation files are present, avoiding doing `i18n:get-translations` previously.

It adds a step to the `dev` command to generate the locales list. The downside of this approach is that it modifies the `valid-locales.json` file and we will need to be alert in PR changes, but do we really need to track it? Looks like it is actually not used.

## Testing Instructions

1. Delete any translation file other than the `en.json`
2. Run the project in this branch: `npm run dev`
3. See no errors.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by zackkrida — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
   "author": "sebworks",
   "private": true,
   "scripts": {
-    "dev": "cross-env ENABLE_AUDIO=true nuxt",
+    "dev": "node src/locales/scripts/get-validated-locales && cross-env ENABLE_AUDIO=true nuxt",
```
Comment (dhruvkb) [inline package.json:15]: We could use the `get-translations` script; or write a different one to just get the validated locales and call it?
```suggestion
    "dev": "npm run i18n:get-translations && cross-env ENABLE_AUDIO=true nuxt",
```

**Review by dhruvkb — APPROVED** [review]
LGTM! Not sure why CI is failing though.

**Review by sarayourfriend — APPROVED** [review]
LGTM. +1 to what Dhruv said. Super great to not track generated files 🙂

**Comment by krysal** [comment]
The `/src/locales/scripts/valid-locales.json` file is used in `nuxt.config.js` so it's critical that the file is present to run the tests. Adding the command to get translations solves it. Do you want to take a second look at this @zackkrida?

File: package.json [hunk 2]
```diff
@@ -12,14 +12,14 @@
...
-    "dev": "node src/locales/scripts/get-validated-locales && cross-env ENABLE_AUDIO=true nuxt",
+    "dev": "npm run i18n:get-translations && cross-env ENABLE_AUDIO=true nuxt",
     "build": "npm run i18n:get-translations && nuxt build",
     "generate": "nuxt generate",
     "start": "nuxt start",
     "storybook": "nuxt storybook",
     "test": "jest",
     "test:watch": "jest --collectCoverage=false --watch",
-    "test:unit": "jest --collectCoverage=false",
+    "test:unit": "npm run i18n:get-translations && jest --collectCoverage=false",
```
Comment (sarayourfriend) [inline package.json:22]: Should this be added to all the `test*` scripts?

Comment (krysal) [reply package.json:22]: Not sure it is necessary as this was sufficient to pass tests in the GH action and adding it to `dev` makes it more likely to be local. I didn't want to overcrowd this file but keeping uniformity sounds like a good reason to add it to the other commands as well :)

**Comment by sarayourfriend** [comment]
If that file is a pre-requisite for running the tests, even locally, then we should probably add it to the test scripts.

I wonder if there's a way to just add it to the jest setup though instead of having to duplicate it across all the NPM scripts.

Which brings up the other option, of using `just` instead of npm scripts as `just` has proper dependency support for each recipe and cross-environment ENV variables built in (we wouldn't need `cross-env`) and much easier support for overriding variables (like right now to run the app locally without audio you need to modify the `package.json`, with `just` we could just make it an argument).

But that's controversial... idk many frontend projects (if any) that don't use `npm` scripts.

**Comment by krysal** [comment]
Thinking again, it makes more sense to add the get-translation step in the Github action! And keep the `package.json` lean, in my view. Do you think the same?

**Comment by krysal** [comment]
I remembered why I had chosen to use just the `get-validated-locales` node script instead of the npm `i18n:get-translations`... Network requests are slow! and doing it whitin every `dev` script run is a pain in slow connections. I'm changing that back.

**Comment by sarayourfriend** [comment]
> I remembered why I had chosen to use just the get-validated-locales node script instead of the npm i18n:get-translations... Network requests are slow! and doing it whitin every dev script run is a pain in slow connections. I'm changing that back.

💯

**Comment by zackkrida** [comment]
@krysal I would add a `"i18n:get-valid-locales": "node src/locales/scripts/get-validated-locales"` script, and add _that_ to the `dev` script, then get this merged 👍

**Comment by krysal** [comment]
Done! Thanks for your suggestions pals 😄

**Pull request merged**